src/wagtail_ai/ai/__init__.py

```diff
@@ -0,0 +1,145 @@
+from collections.abc import Mapping
+from dataclasses import dataclass
+from typing import Any, NotRequired, Required, TypedDict, cast
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.module_loading import import_string
+
+from .. import tokens
+from ..text_splitters.base import BaseTextSplitterLength
+from ..text_splitters.langchain import LangchainRecursiveCharacterTextSplitter
+from ..text_splitters.length import NaiveTextSplitterLength
+from ..types import TextSplitterProtocol
+from .base import AIBackend, ChatModelConfig
+
+
+class InvalidAIBackendError(ImproperlyConfigured):
+    def __init__(self, alias):
+        super().__init__(f"Invalid AI backend: {alias}")
+
+
+class TextSplitterSettings(TypedDict):
+    TEXT_SPLITTER_CLASS: NotRequired[str | None]
+    TEXT_SPLITTER_LENGTH_CLASS: NotRequired[str | None]
+
+
+class ChatModelSettings(TypedDict):
+    ID: Required[str]
+    TOKEN_LIMIT: NotRequired[int | None]
+    TEXT_SPLITTING: NotRequired[TextSplitterSettings | None]
+    EXTRA: NotRequired[Mapping[str, Any] | None]
+
+
+class AIBackendSettings(TypedDict):
+    CHAT_MODEL: Required[ChatModelSettings]
+    CLASS: Required[str]
+    CONFIG: NotRequired[Any]
+
+
+def get_ai_backends_settings() -> Mapping[str, AIBackendSettings]:
+    try:
+        return settings.WAGTAIL_AI["BACKENDS"]
```

Comment:
**question:** With this setting renamed, do we want to add a warning if the old setting is detected?

Reply:
I did not know wagtail-ai is used but this is a good call...